stories/QueryRuleCustomData.stories.js

```diff
@@ -0,0 +1,122 @@
+import { previewWrapper } from './utils';
+import { storiesOf } from '@storybook/vue';
+
+storiesOf('ais-query-rule-custom-data', module)
+  .addDecorator(
+    previewWrapper({
+      indexName: 'instant_search_movies',
+      filters: '<ais-refinement-list attribute="genre" />',
+      hits: `
+      <ol slot-scope="{ items }" class="playground-hits">
+        <li
+          v-for="item in items"
+          :key="item.objectID"
+          class="playground-hits-item"
+        >
+          <div
+            class="playground-hits-image"
+            :style="{ backgroundImage: 'url(' + item.image + ')' }"
+          />
+          <article>
+            <header>
+              <strong><ais-highlight attribute="title" :hit="item"/></strong>
+            </header>
+          </article>
+        </li>
+      </ol>
+      `,
+    })
+  )
+  .add('simple usage', () => ({
+    template: `
+    <div>
+      <p>
+        Type <q>music</q> and a banner will appear.
+      </p>
+      <ais-query-rule-custom-data>
+        <template slot="item" slot-scope="{ item }">
+          <h2>{{ item.title }}</h2>
+          <a :href="item.link">
+            <img
+              :src="item.banner"
+              :alt="item.title"
+              :style="{ width: '100%' }"
+            />
+          </a>
+        </template>
+      </ais-query-rule-custom-data>
+    </div>
+    `,
+  }))
+  .add('with default banner', () => ({
+    template: `
+    <div>
+      <p>
+        Kill Bill appears whenever no other results are promoted. Type
+        <q>music</q> to see another movie promoted.
+      </p>
+      <ais-query-rule-custom-data :transform-items="transformItems">
+        <template slot="item" slot-scope="{ item }">
+          <h2>{{ item.title }}</h2>
+          <a :href="item.link">
+            <img
+              :src="item.banner"
+              :alt="item.title"
+              :style="{ width: '100%' }"
+            />
+          </a>
+        </template>
+      </ais-query-rule-custom-data>
+    </div>
+    `,
+    data() {
+      return {
+        transformItems: items => {
+          if (items.length > 0) {
+            return [items[0]];
```

Comment:
Is there a reason to only pick the first item rather than the full array?

Reply:
This is an example so you always only show one banner, which might be the case for most designs